### PR TITLE
Update non res with dormancy features

### PR DESCRIPTION
**Pull request by MarionJHolloway**
# Description
This PR updates the list of features in the non-res with dormancy features job to include time open. Unit tests and validation were also added.
A small change was made to the clean CQC location job so that the imputed registration date is now in a date format instead of string format.

# How to test
Unit tests passing
Run successfully in branch: https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:344210435447:execution:update-non-res-with-dormancy-f-Ind-CQC-Filled-Post-Estimates-Pipeline:f08b4719-d9f5-4b95-be50-50a2ce05bffc
Run in branch after PR feedback: https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:344210435447:execution:update-non-res-with-dormancy-f-Ind-CQC-Filled-Post-Estimates-Pipeline:994ce96a-a6c0-4102-b44d-333c58790496

# Developer checklist
- [x] Unit test
- [x] Linked to Trello ticket: https://trello.com/c/zwIJq1KL/611-epic-1-update-features-job-for-non-residential-ascwds-with-dormancy-model-must
- [x] Documentation up to date

### CODE REVIEW

File: utils/validation/validation_rules/non_res_ascwds_inc_dormancy_ind_cqc_features_validation_rules.py [hunk 1]
```diff
@@ -3,16 +3,43 @@
...
         RuleName.index_columns: [
             IndCqcColumns.location_id,
             IndCqcColumns.cqc_location_import_date,
         ],
+        RuleName.min_values: {
+            IndCqcColumns.ascwds_filled_posts_dedup_clean: 1.0,
+        },
+        RuleName.max_values: {
+            IndCqcColumns.ascwds_filled_posts_dedup_clean: 3000.0,
+        },
```
Comment (garypolzin) [inline utils/validation/validation_rules/non_res_ascwds_inc_dormancy_ind_cqc_features_validation_rules.py:34]: add validations for time open too?

shouldn't be negative, no nulls

File: utils/features/helper.py [hunk 2]
```diff
@@ -55,3 +55,26 @@ def add_date_diff_into_df(
         new_col_name, F.datediff(F.lit(max_d), F.col(import_date_col))
     )
     return loc_df
+
+
+def add_time_open_into_df(df: DataFrame, new_col_name: str) -> DataFrame:
```
Comment (garypolzin) [inline utils/features/helper.py:60]: do you think this needs to be more specifically named, such as
number_of_days_between_registration_date_and_file_date / length_of_time_registered_for
(it's the date of registration, not 'open' - could have been open longer but as a non-cqc place)

File: utils/column_values/categorical_columns_by_dataset.py [hunk 3]
```diff
@@ -154,10 +154,13 @@ class FeatureEngineeringCategoricalValues:
     current_region_column_values = Region(IndCQC.current_region)
     services_column_values = Services(IndCQC.gac_service_types)
     current_rui_column_values = RUI(IndCQC.current_rural_urban_indicator_2011)
-    dormancy_column_values = Dormancy(IndCQC.dormancy, contains_null_values=True)
+    dormancy_column_values = Dormancy(IndCQC.dormancy)
     care_home_column_values = CareHome(
```
Comment (garypolzin) [inline utils/column_values/categorical_columns_by_dataset.py:158]: does this one need renaming to match the format in the one below:
care_home_column_care_home_values

**Review by garypolzin — CHANGES_REQUESTED** [review]
looks great, couple of comments but mostly optional

File: utils/features/helper.py (hunk 2 above)
Comment (MarionJHolloway) [reply utils/features/helper.py:60]: I went for time registered because it seemed more accurate but still concise

**Review by garypolzin — APPROVED** [review]

**Pull request merged**